rust/src/measurements/make_private_expr/expr_report_noisy_max/mod.rs

```diff
@@ -101,6 +101,26 @@ where
         );
     }
 
+    use DataType::*;
+    if let DataType::Array(dtype, _) = middle_domain.column.dtype() {
```

Comment:
```suggestion
    if let DataType::Array(dtype, _) != middle_domain.column.dtype() {
```
Slight preference to flatten the nested ifs, for readability: First one check, and fail if needed, and then a second check, and another fail.

Reply:
Switched to a let-else binding.